modules/imgproc/src/canny.cpp

```diff
@@ -318,6 +316,8 @@ class parallelCanny : public ParallelLoopBody
         // In sobel transform we calculate ksize2 extra lines for the first and last rows of each slice
         // because IPPDerivSobel expects only isolated ROIs, in contrast with the opencv version which
```

Comment:
Note that current integration of IPP in Sobel supports in memory border and this part probably can be more efficient.